src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Tree/Dependencies/DependencyNodeIdTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies
+{
+    [ProjectSystemTrait]
+    public class DependencyNodeIdTests
+    {
+        [Fact]
+        public void DependencyNodeId_Constructor()
+        {
+            Assert.Throws<ArgumentNullException>("providerType", () => {
+                var id = new DependencyNodeId(null, null, null, null);
+            });
+
+            Assert.Throws<ArgumentException>("providerType", () => {
+                var id = new DependencyNodeId("", null, null, null);
+            });
+        }
+
+        [Theory]
+        [InlineData("MyProviderType", "MyItemSpec", "MyItemType", "MyUniqueToken",
+                    "file:///[MyProviderType;MyItemSpec;MyItemType;MyUniqueToken]")]
+        [InlineData("MyProviderType", "MyItemSpec", "MyItemType", null,
+                    "file:///[MyProviderType;MyItemSpec;MyItemType]")]
+        [InlineData("MyProviderType", "MyItemSpec", "", null,
+                    "file:///[MyProviderType;MyItemSpec]")]
+        [InlineData("MyProviderType", null, null, null,
+                    "file:///[MyProviderType]")]
```

Comment:
Good point, i adde more cases and made DependencyNodeId serialize properties at the same positions always.

---
In reply to: [74356455](https://github.com/dotnet/roslyn-project-system/pull/349#discussion_r74356455) [](ancestors = 74356455)